test/provisioner/provision_azure_initializer.go

```diff
@@ -0,0 +1,165 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package provisioner
+
+import (
+	"errors"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/containerservice/mgmt/containerservice"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	initAzureLogger()
+}
+
+func initAzureLogger() {
+	level := os.Getenv("LOG_LEVEL")
+	switch level {
+	case "trace":
+		log.SetLevel(log.TraceLevel)
+	case "debug":
+		log.SetLevel(log.DebugLevel)
+	case "info":
+		log.SetLevel(log.InfoLevel)
+	case "warn":
+		log.SetLevel(log.WarnLevel)
+	case "error":
+		log.SetLevel(log.ErrorLevel)
+	case "fatal":
+		log.SetLevel(log.FatalLevel)
+	case "panic":
+		log.SetLevel(log.PanicLevel)
+	default:
+		log.SetLevel(log.InfoLevel)
+	}
+}
+
+type AzureProperties struct {
+	ResourceGroup     *resources.Group
+	CloudProvider     string
+	SubscriptionID    string
+	ClientID          string
+	ClientSecret      string
+	TenantID          string
+	ResourceGroupName string
+	ClusterName       string
+	Location          string
+	SshKeyID          string
+	SubnetName        string
+	VnetName          string
+	SubnetID          string
+	ImageID           string
+
+	InstanceSize string
+	NodeName     string
+	OsType       string
+
+	ResourceGroupClient  *resources.GroupsClient
+	ManagedclusterClient *containerservice.ManagedClustersClient
+	ManagedvnetClient    *network.VirtualNetworksClient
+	ManagedsubnetClient  *network.SubnetsClient
+	ManagedaksClient     *containerservice.ManagedClustersClient
+	ManagedvmClient      *compute.VirtualMachinesClient
+}
+
+var AzureProps = &AzureProperties{}
+
+func initazureProperties(properties map[string]string) error {
+	log.Trace("initazureProperties()")
+	AzureProps = &AzureProperties{
+		SubscriptionID:    properties["AZURE_SUBSCRIPTION_ID"],
+		ClientID:          properties["AZURE_CLIENT_ID"],
+		ClientSecret:      properties["AZURE_CLIENT_SECRET"],
+		TenantID:          properties["AZURE_TENANT_ID"],
+		ResourceGroupName: properties["RESOURCE_GROUP_NAME"],
+		ClusterName:       properties["CLUSTER_NAME"],
+		Location:          properties["LOCATION"],
+		SshKeyID:          properties["SSH_KEY_ID"],
+		CloudProvider:     properties["CLOUD_PROVIDER"],
+	}
+
+	AzureProps.VnetName = AzureProps.ClusterName + "_vnet"
+	AzureProps.SubnetName = AzureProps.ClusterName + "_subnet"
+	AzureProps.InstanceSize = "Standard_D8as_v5"
+	AzureProps.NodeName = "caaaks"
+	AzureProps.OsType = "Ubuntu"
+
+	log.Debugf("%+v", AzureProps)
+
+	if len(AzureProps.SubscriptionID) <= 0 {
```

Comment:
For a string even this is an easy way to check if it empty!

```suggestion
	if AzureProps.SubscriptionID == "" {
```